### PR TITLE
Sort Public Subissions By User Langauge

### PR DESCRIPTION
**Pull request by MrinallU**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by MrinallU** [comment]
This was just a small feature that was bugging me, but when I look at public submissions, I usually have to scroll through a lot of c++ solutions before reaching a java sol. This just lists the solutions that match the user's selected language.

**Comment by SansPapyrus683** [comment]
is it just me or are all the build tests failing for no reason

**Comment by MrinallU** [comment]
> is it just me or are all the build tests failing for no reason

Yup, not sure why.

File: src/components/ProblemSolutions.tsx [hunk 1]
```diff
@@ -22,9 +22,18 @@ export default function ProblemSolutions({
...
-  const { firebaseUser } = useContext(UserDataContext);
+  const { firebaseUser, lang } = useContext(UserDataContext);
   const { signIn } = React.useContext(SignInContext);
   const canModerate = useUserPermissions().canModerate;
+  const langArr = ['cpp', 'java', 'py'];
+  langArr.sort(function (first, second) {
+    if (first == lang && second != lang) {
+      return -1;
+    } else if (first != lang && second == lang) {
+      return 1;
```
Comment (nathangong) [inline src/components/ProblemSolutions.tsx:33]:
```suggestion
    if (first === lang && second !== lang) {
      return -1;
    } else if (first !== lang && second === lang) {
      return 1;
```

**Review by nathangong — APPROVED** [review]
didn't test but code looks good

**Review by maggie-j-liu — APPROVED** [review]
looks good, thank you! in the future we could add a dropdown to filter by language 👀

**Pull request merged**